src/containers/ModelPanel/selectors.ts

```diff
@@ -32,35 +36,37 @@ export const convertUITreeDataToSelectNone = createSelector(
     }
 );
 
-export const getCheckboxAllIsIntermediate = createSelector(
+// Determines if the shared checkbox should be partially checked (in "indeterminate" state)
+export const getIsSharedCheckboxIndeterminate = createSelector(
     [getUiDisplayDataTree, getAgentVisibilityMap],
-    (treeData, visibleAgents) => {
-        let childrenIntermediate = false;
+    (
+        allAgents: AgentDisplayNode[],
+        agentVisibilityMap: VisibilitySelectionMap
+    ): boolean => {
+        if (isEmpty(agentVisibilityMap)) return false;
+
+        let isAgentPartiallyVisible = false;
+        let numInvisibleAgents = 0;
+
+        // Loop through each agent and count how many are invisible. If an agent is partially
+        // visible, break out of loop and just return true
+        for (let i = 0; i < allAgents.length; i++) {
+            const agent = allAgents[i];
+            const visibleStates = agentVisibilityMap[agent.key];
```

Comment:
I'm surprised that typeScript doesn't tell you `object this possibility undefined here` since it's using the key from one map to access another. Did giving types to the two inputs remove that?

Reply:
I think adding `if (isEmpty(agentVisibilityMap)) return false;` in line 46 took care of that

Reply:
Oh wait, I get what you're asking now. I didn't get an error like that, maybe because I put in the types for the inputs :)

Reply:
Maybe I should add a check here just in case `agentVisibilityMap` for some reason doesn't have `agent.key` as a key?

Reply:
Added a check for when `visibleStates` is undefined